docs/concepts/custom-ui-headless-api.mdx

```diff
@@ -0,0 +1,662 @@
+---
+id: custom-ui-headless-api
+title: Integration Basics
+sidebar_label: Integration Basics
+toc_max_heading_level: 4
+---
+
+```mdx-code-block
+import Tabs from "@theme/Tabs"
+import TabItem from "@theme/TabItem"
+import CodeBlock from "@theme/CodeBlock"
+```
+
+Ory provides headless APIs for ease of integration. This ensures that Ory provides compatibility with many software stacks across
+different programming languages.
+
+This documents' focus is to provide concrete examples on custom UI integration. Flows are the fundamental of Ory's self-service
+APIs and are used to accomplish a variety of goals, such as `login`, `registration`, `settings`, `verification` and `recovery`.
+
+The self-service `flows` are standardized so that it follows the same steps regardless of which `flow` is integrated.
+
+A general overview of the flow can be summed up in two steps:
+
+1. Get or create the `flow` data, e.g. login, registration etc.
+2. Submit the `flow` with user data (e.g. username and password)
```

Comment:
I would make this more verbose:

```suggestion
1. Create the flow with the appropriate purpose and application.
2. Use the flow data to render the UI.
3. Submit the flow with user data.
4. Potentially handle flow errors like invalid values by going back to 2.
5. Handle successful submission.
```

The use-cases can become very complex, but first explaining the "standard" use-case should help people to wrap their head around the concepts.